### ISSUE TITLE
Strange behaviour in mesh rasterization (losing faces)

### ISSUE THREAD

**Issue by weders** [facebookresearch/pytorch3d#1221]
I am currently using Pytorch3D to render synthetic depth maps from a bunch of bounding boxes around objects and use them to compute occlusion. 

The code I use for doing that is the following:

```
# prepare camera rendering        
camera_render = get_pytorch3d_camera(np.linalg.inv(pose), intrinsics).to(device)
rasterizer = MeshRasterizer(cameras=camera_render, raster_settings=raster_settings).to(device)

fragments = rasterizer(meshes_boxes)
face_ids = fragments.pix_to_face[0, :, :, 0].cpu().detach().numpy()
depth_rendered = fragments.zbuf[0, :, :, 0].cpu().detach().numpy()
```

From this code, I expect depth maps as the following (where the order is (real depth map, real image, synthetic depth map, occlusion mask):

![image](https://user-images.githubusercontent.com/17823018/172367285-43c91570-e625-4eda-828a-b6b3262df90e.png)
![image](https://user-images.githubusercontent.com/17823018/172367312-99670e33-a5ea-4634-a170-9faf6fa70e96.png)

However, a few frames later it seems like the front faces are lost and only the back faces are rendered. Some examples you can see below:

![image](https://user-images.githubusercontent.com/17823018/172367631-ba32aae8-930c-45b8-b9b0-803891d20a49.png)

Then one of the front faces appears again

![image](https://user-images.githubusercontent.com/17823018/172367701-c42f45a5-7c22-44b0-9da1-fbf87b469539.png)

And the wrongly rendered backface is also lost later

![image](https://user-images.githubusercontent.com/17823018/172367768-6dec194a-3ae8-4f63-9711-03e1a5f5f569.png)

Any idea what could cause this unexpected behaviour? To me it seems like some faces are lost. However, when I check the shape of mesh._faces_packed, it does not change.

**Comment by gkioxari** [comment]
Could you provide data and a *compact* code snippet so that I can reproduce this? I think your settings might be off but I will be able to tell you once I had code + data!
